### PR TITLE
chore: improve auth debug logging

### PR DESCRIPTION
**Pull request by PeterSchafer**
- [x] Follows [CONTRIBUTING](https://github.com/snyk/snyk/blob/master/CONTRIBUTING.md) rules

#### What does this PR do?
Improve debug logging around authorization usage by using the active authenticator.

### CODE REVIEW

File: cliv2/cmd/cliv2/main.go [hunk 1]
```diff
@@ -233,34 +235,54 @@ func displayError(err error) {
...
+func logHeaderAuthorizationInfo(config configuration.Configuration, networkAccess networking.NetworkAccess) (string, string) {
+	oauthEnabled := "Disabled"
+	authorization := ""
+
+	apiRequest := &http.Request{
+		URL:    config.GetUrl(configuration.API_URL),
+		Header: http.Header{},
+	}
+	err := networkAccess.GetAuthenticator().AddAuthenticationHeader(apiRequest)
+	if err == nil {
```
Comment (bastiandoetsch) [inline cliv2/cmd/cliv2/main.go:247]: how about inverting this, as the handling gets even into deeper branching further down.

File: cliv2/cmd/cliv2/main.go [hunk 2]
```diff
@@ -233,34 +235,54 @@ func displayError(err error) {
...
+		tokenShaSum := []byte{}
+		tokenDetails := ""
+
+		authHeader := apiRequest.Header.Get("Authorization")
+		splitHeader := strings.Split(authHeader, " ")
+		if len(splitHeader) == 2 {
```
Comment (bastiandoetsch) [inline cliv2/cmd/cliv2/main.go:253]: inverting could be considered here, too

File: cliv2/cmd/cliv2/main.go [hunk 3]
```diff
@@ -233,34 +235,54 @@ func displayError(err error) {
...
+			token := splitHeader[1]
+			temp := sha256.Sum256([]byte(token))
+			tokenShaSum = temp[0:16]
+			tokenDetails = fmt.Sprintf(" (type=%s)", tokenType)
+
+			if config.GetBool(configuration.OAUTH_AUTH_ENABLED) {
+				oauthEnabled = "Enabled"
+				token, err := auth.GetOAuthToken(config)
+				if token != nil && err == nil {
+					tokenDetails = fmt.Sprintf(" (type=oauth; expriy=%v)", token.Expiry.UTC())
```
Comment (bastiandoetsch) [inline cliv2/cmd/cliv2/main.go:264]: `expriy` is probably meant to mean `expiry` :)

**Review by bastiandoetsch — APPROVED** [review]

File: cliv2/cmd/cliv2/main.go (hunk 1 above)
Comment (PeterSchafer) [reply cliv2/cmd/cliv2/main.go]: done!

**Pull request merged**